discovery-infra/test_infra/logs_utils.py

```diff
@@ -32,18 +33,14 @@ def verify_logs_uploaded(cluster_tar_path, expected_min_log_num, installation_su
                         _verify_oc_logs_uploaded(os.path.join(tempdir, gz))
 
 
-def wait_and_verify_oc_logs_uploaded(cluster, cluster_tar_path, num_retries=NUM_OF_RETRIES):
-    for retry in range(num_retries):
-        try:
-            time.sleep(OC_DOWNLOAD_LOGS_INTERVAL)
-            cluster.download_installation_logs(cluster_tar_path)
-            assert os.path.exists(cluster_tar_path), f"{cluster_tar_path} doesn't exist"
-            _verify_oc_logs_uploaded(cluster_tar_path)
-            return
-        except AssertionError as err:
-            logging.info(f'attempt {retry + 1} to download failed with error {str(err)}')
-
-    assert False, "oc logs were not uploaded"
+def wait_and_verify_oc_logs_uploaded(cluster, cluster_tar_path):
+    try:
+        cluster.wait_for_logs_complete(timeout=OC_DOWNLOAD_LOGS_TIMEOUT, interval=OC_DOWNLOAD_LOGS_INTERVAL, host_only=False)
+        cluster.download_installation_logs(cluster_tar_path)
+        assert os.path.exists(cluster_tar_path), f"{cluster_tar_path} doesn't exist"
+        _verify_oc_logs_uploaded(cluster_tar_path)
+    except BaseException as err:
+        assert False, f'oc logs were not uploaded {str(err)}'
```

Comment:
Should treat wrapping exceptions like [this](https://docs.python.org/3/library/exceptions.html#built-in-exceptions)

```suggestion
    except BaseException as err:
        raise ConnectionError("oc logs were not uploaded") from err
```